stock_inventory_revaluation/models/account_move.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2022 ForgeFlow S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    stock_revaluation_count = fields.Integer(compute="_compute_stock_revaluation_count")
+    show_create_stock_revaluation_button = fields.Boolean(
+        compute="_compute_show_create_stock_revaluation_button"
+    )
+
+    def button_revaluate(self):
+        """Create a revaluation record associated to this bill"""
+        self.ensure_one()
+        inventory_revaluation = self.env["stock.inventory.revaluation"].create(
+            {
+                "vendor_bill_id": self.id,
+            }
+        )
+        action = self.env["ir.actions.actions"]._for_xml_id(
+            "stock_inventory_revaluation.action_stock_inventory_revaluation"
+        )
+        return dict(
+            action,
+            view_mode="form",
+            res_id=inventory_revaluation.id,
+            views=[(False, "form")],
+        )
+
+    def action_view_inventory_revaluation_lines(self):
+        self.ensure_one()
+        action = self.env["ir.actions.actions"]._for_xml_id(
+            "stock_inventory_revaluation.action_stock_inventory_revaluation"
+        )
+        revaluations = self.env["stock.inventory.revaluation"].search(
+            [("vendor_bill_id", "=", self.id)]
+        )
+        domain = [("id", "in", revaluations.ids)]
+        context = dict(self.env.context, default_vendor_bill_id=self.id)
+        views = [
+            (
+                self.env.ref(
+                    "stock_inventory_revaluation.view_stock_inventory_revaluation_tree"
+                ).id,
+                "tree",
+            ),
+            (False, "form"),
+        ]
+        return dict(action, domain=domain, context=context, views=views)
+
+    def _get_account_move_line_price_unit(self, currency, product_id):
+        """get the price unit and qty of the move lines for a given product"""
+        self.ensure_one()
+        return sum(
+            [
+                currency._convert(
+                    invl.price_subtotal,
+                    invl.company_currency_id,
+                    invl.company_id,
+                    invl.move_id.date,
+                )
+                for invl in self.line_ids.filtered(lambda l: l.product_id == product_id)
+            ]
+        ), sum(
+            [
+                invl.quantity
+                for invl in self.line_ids.filtered(lambda l: l.product_id == product_id)
+            ]
+        )
+
+    def _compute_stock_revaluation_count(self):
+        for rec in self:
+            revs = self.env["stock.inventory.revaluation"].search(
+                [("vendor_bill_id", "=", rec.id)]
+            )
+            rec.stock_revaluation_count = len(revs)
```

Comment:
can use `search_count `or `read_group`(out of loop) is more efficient only for count

Reply:
Thanks for the suggestion! :) Changes applied